### ISSUE TITLE
How to reproduce Figure 6 (muscle activation)?

### ISSUE THREAD

**Issue by mulkkyul** [lsw9021/MASS#6]
Hi there,
Thanks for sharing the code!

I've been working with your paper and I'd like to reconstruct Figure 6 in your paper.

After training, the model can walk and I somehow obtained 10 x 284 muscle activation for each step. 

I tried to reproduce Figure 6 (muscle activation), but then I'm kinda lost. 
There are 284 muscles, and I don't know how to use/choose them to have a plot like Figure 6.

Could you give me some tips?
Thanks!

**Comment by lsw9021** [comment]
Hi Mr. Jungsik,

Figure 6 in the paper is called eight phases of walking. As for the data, I've collected the muscle activation levels for 40 cycles. As we use line approximated muscle model, one muscle can be divided into several line-segments. For example, the Bicep Brachialis has two heads (the long head and the short head) to represent its geometry. In that case, you have to sum the two line muscle and average them to correctly observe the activations.

In our case, there are two things that matter when collecting the data. One is that the eight phases of walking are not temporally fixed. The elapsed time between Initial Contact and Loading Response is shorter than the time between Loading Response and Mid Stance. You should warp the time domain to get the data. To do so, we additionally observe the foot positions and check whether the foot is in contact or not. The other is that normalization. Every EMG data don't have actual magnitude, but rather provide a ratio which describes the patterns of muscle activation. We normalize every muscle activation level using maximum activation level.

Please check the above and plot the muscle activation. Please let me know if you have some problem.

**Comment by mulkkyul** [comment]
Thank you for the answer.
I think I'm getting it but I still have one question about line segments.

There're same names in muscle24.xml. For example,

there are L_Gluteus_Maximus, L_Gluteus_Maximus1, L_Gluteus_Maximus2, L_Gluteus_Maximus3 and L_Gluteus_Maximus4. 

Are these line segments? 
What's the difference between the one with number and the ones without?

and do I need all of them to plot the muscle activation? (for instance, getting the average of **L_Gluteus_Maximus, L_Gluteus_Maximus1, L_Gluteus_Maximus2, L_Gluteus_Maximus3 and L_Gluteus_Maximus4** to depict the muscle activation of **L_Gluteus_Maximus**?

I'm new to this muscle-skeletal simulation, so please excuse my silly questions.
Thank you very much!

Cheers,

**Comment by todayThursday** [comment]
> 您好，感谢您分享代码！
> 
> 我一直在研究您的论文，我想在您的论文中重建图 6。
> 
> 训练后，模型可以走路了，我不知何故每走一步获得了 10 x 284 的肌肉激活。
> 
> 我试图重现图 6（肌肉激活），但后来我有点迷失了。有 284 块肌肉，我不知道如何使用/选择它们来获得如图 6 所示的图。
> 
> 您能给我一些提示吗？ 谢谢！

I have been training the running model recently, but no matter how to modify reward_param, the training effect is not good, I want to know if you have solved this problem, but on the other hand, I think the weight parameter of the reward function does not affect the effect of the training model, in the code, The reward function only considers the position error and the velocity error of the joint, and scales these two with the error of the end-effector. Logically, if we only consider the similarity of the motion, and do not need to add other effects, just the results of the training of the neural network, then why does the run model not have good results no matter how it is felt
